src/python/grpcio/grpc/beta/_client_adaptations.py

```diff
@@ -159,9 +159,11 @@ def time_remaining(self):
     return self._call.time_remaining()
 
   def add_abortion_callback(self, abortion_callback):
-    registered = self._call.add_callback(
-        lambda: abortion_callback(_abortion(self._call)))
-    return None if registered else _abortion(self._call)
+    def done_callback(unused_arg):
+      if self.code() is not grpc.StatusCode.OK:
+        return abortion_callback(_abortion(self._call))
+    registered = self._future.add_done_callback(done_callback)
```

Comment:
`self._future` may not always be non-`None`. Should what's called here be `self._call.add_callback`?

... and it looks like `add_callback` isn't implemented on the client-side. Except that it is, under the name `add_cancellation_callback`. So that should be an easy fix.

... so it looks like the way we're mixing `six` and `abc` isn't working to provide the guarantees that `abc` is supposed to provide. I'll look into that.